src/Sparrow/Compression/LZ4.cs

```diff
@@ -79,6 +79,35 @@ protected struct LZ4_stream_t_internal
             public uint initCheck;
         }
 
+        public long Encode64LongBuffer(
+            byte* input,
+            byte* output,
+            long inputLength,
+            long outputLength,
+            int acceleration = ACCELERATION_DEFAULT)
+        {
+            if (inputLength < int.MaxValue && outputLength < int.MaxValue)
```

Comment:
At any rate, we are probably better at compressing at 1GB size, to ensure that we are away from the limit.